test/functional/mempool_package_onemore.py

```diff
@@ -55,14 +55,23 @@ def run_test(self):
         assert_raises_rpc_error(-26, "too-long-mempool-chain, too many descendants", self.chain_tx, [chain[0], second_chain])
         # ...especially if its > 40k weight
         assert_raises_rpc_error(-26, "too-long-mempool-chain, too many descendants", self.chain_tx, [chain[0]], num_outputs=350)
+        # ...even if it's submitted with other transactions
+        replaceable_tx = self.wallet.create_self_transfer_multi(utxos_to_spend=[chain[0]])
+        txns = [replaceable_tx["tx"], self.wallet.create_self_transfer_multi(utxos_to_spend=replaceable_tx["new_utxos"])["tx"]]
+        txns_hex = [tx.serialize().hex() for tx in txns]
+        assert_equal(self.nodes[0].testmempoolaccept(txns_hex)[0]["reject-reason"], "too-long-mempool-chain")
+        pkg_result = self.nodes[0].submitpackage(txns_hex)
+        assert "too-long-mempool-chain" in pkg_result["tx-results"][txns[0].getwtxid()]["error"]
+        assert_equal(pkg_result["tx-results"][txns[1].getwtxid()]["error"], "bad-txns-inputs-missingorspent")
         # But not if it chains directly off the first transaction
-        replacable_tx = self.wallet.send_self_transfer_multi(from_node=self.nodes[0], utxos_to_spend=[chain[0]])['tx']
+        self.nodes[0].sendrawtransaction(replaceable_tx["hex"])
         # and the second chain should work just fine
         self.chain_tx([second_chain])
 
         # Make sure we can RBF the chain which used our carve-out rule
-        replacable_tx.vout[0].nValue -= 1000000
-        self.nodes[0].sendrawtransaction(replacable_tx.serialize().hex())
+        replacement_tx = replaceable_tx["tx"]
+        replacement_tx.vout[0].nValue -= 1000000
+        self.nodes[0].sendrawtransaction(replacement_tx.serialize().hex())
 
```

Comment:
Maybe test that Package RBF carveout is not supported?
```suggestion

        # But We can not package RBF the chain which used our carveout rule
        replaceable_tx_conflict = self.wallet.create_self_transfer_multi(utxos_to_spend=[chain[0]])
        txns_conflict = [replaceable_tx_conflict["tx"], self.wallet.create_self_transfer_multi(utxos_to_spend=replaceable_tx_conflict["new_utxos"])["tx"]]
        txns_conflict_hex = [tx.serialize().hex() for tx in txns_conflict]
        assert_equal(self.nodes[0].testmempoolaccept(txns_conflict_hex)[0]["reject-reason"], "bip125-replacement-disallowed")
        pkg_result = self.nodes[0].submitpackage(txns_conflict_hex)
        assert "too-long-mempool-chain" in pkg_result["tx-results"][txns_conflict[0].getwtxid()]["error"]
        assert_equal(pkg_result["tx-results"][txns_conflict[1].getwtxid()]["error"], "bad-txns-inputs-missingorspent")
```

Reply:
would need to do a bit more work to make sure it actually would be attempting a package rbf. I think we can rely on our conflict topology check tests to ensure this, at least for default mempool chains?

Reply:
I learned that we can RBF a transaction that gets into the mempool using carveout. I wanted to test that you wouldn't  RBF  that transaction using  a package.

Even if the arrangement attempts to RBF or not, I think the error in the suggestion is how it would fail. We would set `m_allow_carveout` to false while evaluating the first parent transaction and fail its evaluation due to a `TX_MEMPOOL_POLICY` violation of `too-long-mempool-chain`. hence, all the descendants would fail due to missing input.

I just think this is a "nice-to-have" test, but not necessarily required for this PR.

Reply:
Ahh I see, it's the same code path.

Reply:
oh right, neither of these cases will do a package rbf, as the package has in-mempool ancestors.

It would have to be a scenario where the ancestor limits are set to 1, then a package RBF of size 2 replaces that single tx?

Reply:
Hmm, I think it will always be the same code path that will be hit because the parent transaction (in the cluster size 2 replacement package) will be detected as conflict with the transaction that was carved out (because it has to go through `PreChecks`), and we don't allow package RBF carveout, so the subpackage evaluation will not be executed?

Reply:
Right, *even if* the test would manage to hit the case where it would have considered a package rbf(this case won't), it should be caught in both PreChecks and package mempool checks separately.